spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/query/QueryEnhancerFactory.java

```diff
@@ -37,6 +37,9 @@ public final class QueryEnhancerFactory {
 	private static final boolean hibernatePresent = ClassUtils.isPresent("org.hibernate.query.TypedParameterValue",
 			QueryEnhancerFactory.class.getClassLoader());
 
+	private static final boolean eclipseLinkPresent = ClassUtils
```

Comment:
Any reason to not use `org.eclipse.persistence.jpa.JpaEntityManager`? Generally, why do we not introduce a property on `PersistenceProvider` to check whether it is available on the class path?

Introducing redundant checks results in code that is difficult to maintain.

Reply:
Okay, I crafted a solution to essentially check for the presence of the same class names used to construct the instances of `PersistenceProvider`. I also created an interface, `PresenceDetector`, which I'm not 100% certain we need. Feel free to drop that. It should be visible below.